crates/ra_hir/src/ty/traits.rs

```diff
@@ -125,11 +126,11 @@ fn solution_from_chalk(db: &impl HirDatabase, solution: chalk_solve::Solution) -
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub(crate) struct SolutionVariables(pub Canonical<Vec<Ty>>);
+pub struct SolutionVariables(pub Canonical<Vec<Ty>>);
```

Comment:
Does the visibility need to change here? A quick look at the diff suggests no but I'm probably missing it.

Reply:
Yeah, I also wondered that.... We can always make stuff private later I guess.

Reply:
Adding the `implements` query means `Solution` needs to be pub to avoid private-in-public errors. Not sure if I checked if it's really necessary for `SolutionVariables` as well.